petl/io/xlsx.py

```diff
@@ -110,7 +112,10 @@ def toxlsx(tbl, filename, sheet=None, write_header=True, mode="replace"):
     wb = _load_or_create_workbook(filename, mode, sheet)
     ws = _insert_sheet_on_workbook(mode, sheet, wb)
     if write_header:
-        rows = tbl
+        it = iter(tbl)
+        hdr = next(it)
```

Comment:
We are backed by tests

Reply:
- I was unsure that a few tests could catch all cases but after thinking a bit...
- It looks safe because is writing and not reading
- It looks like a improvement and a behavior fix
- We can revert if some breakage happens